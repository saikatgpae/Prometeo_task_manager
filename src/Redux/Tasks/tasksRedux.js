const initialState = localStorage.getItem('tasks');
// console.log(initialState.filter((task, index) => index !== 3));

const FETCH_TASKS = 'FETCH_TASKS';

// Fetch tasks from local storage if any
const fetchTasks = () => (dispatch) => {
  const tasks = localStorage.getItem('tasks');
  dispatch({
    type: FETCH_TASKS,
    tasks,
  });
};

// eslint-disable-next-line default-param-last
const tasksReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TASKS:
      return [
        JSON.parse(state),
      ];
    default:
      return state;
  }
};

export default tasksReducer;
export { fetchTasks };
