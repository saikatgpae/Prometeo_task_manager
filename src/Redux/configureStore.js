import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './Tasks/tasksRedux';

const rootReducer = combineReducers({
  tasks: tasksReducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
