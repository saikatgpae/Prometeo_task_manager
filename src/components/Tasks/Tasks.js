// /* eslint-disable */
import { useDispatch, useSelector } from 'react-redux';
import './Tasks.css';
import { useEffect } from 'react';
import AddTasks from '../AddTasks/AddTasks';
import Task from '../Task/Task';
import { fetchTasks } from '../../Redux/Tasks/tasksRedux';

export default function Tasks() {
  const dispatch = useDispatch();
  // Call the tasks from Redux
  const tasks = useSelector((state) => state.tasks);
  useEffect(() => {
    if (tasks === null) {
      dispatch(fetchTasks());
    }
  }, [dispatch, tasks]);

  return (
    <>
      <AddTasks />
      <Task tasks={tasks} />
    </>
  );
}
