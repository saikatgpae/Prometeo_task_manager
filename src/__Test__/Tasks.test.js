import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Tasks from '../components/Tasks/Tasks';

const initialState = [
  { complete: false, taskName: 'playing' },
  { complete: false, taskName: 'drawing' },
  { complete: false, taskName: 'Shopping' },
];
const mockStore = configureStore([]);
const store = mockStore(initialState);

test('Tasks render correctly', () => {
  const component = renderer.create(<Provider store={store}><Tasks /></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
