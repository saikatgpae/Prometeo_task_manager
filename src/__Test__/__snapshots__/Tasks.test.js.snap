// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks render correctly 1`] = `
Array [
  <div
    className="m-3"
  >
    <input
      className="m-2 add-input p-2"
      data-testid="add-task-input"
      id="taskName"
      name="taskName"
      onChange={[Function]}
      placeholder="Add a Task"
      type="text"
    />
    <button
      className="btn btn-primary"
      data-testid="add-task-button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>,
  <div
    className="container container-div"
  >
    <div
      className="table-responsive table-div"
    />
  </div>,
]
`;
